arbigent-core/src/main/java/io/github/takahirom/arbigent/ArbigentReplayScriptWriter.kt

```
@@ -0,0 +1,255 @@
+    fun writeAtomically(target: File, text: String, executable: Boolean = false) {
+      val temp = File(target.parentFile, "${target.name}.${ProcessHandle.current().pid()}.${System.nanoTime()}.tmp")
+      try {
+        temp.writeText(text)
+        if (executable) temp.setExecutable(true, false)
+        try {
+          java.nio.file.Files.move(
+            temp.toPath(), target.toPath(),
+            java.nio.file.StandardCopyOption.REPLACE_EXISTING,
+            java.nio.file.StandardCopyOption.ATOMIC_MOVE,
+          )
+        } catch (e: java.io.IOException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)